python/mxnet/optimizer.py

```diff
@@ -332,7 +333,8 @@ def create_state(self, index, weight):
         if self.momentum == 0.0:
             return None
         else:
-            return zeros(weight.shape, weight.context, dtype=weight.dtype)
+            return sparse_zeros(weight.storage_type, weight.shape,
```

Comment:
We discussed this last Friday. In order to do this, we need to move all the util functions supporting sparse ndarrays in ndarray.py into an ndarray util python file to avoid circular import between ndarray.py and sparse_ndarray.py. Quite a few files calling those functions will be affected.